lib/pf/UnifiedApi.pm

```diff
@@ -107,6 +105,10 @@ our @API_V1_ROUTES = (
 
     { controller => 'nodes', collection_v2a => { get => 'latest_locationlog_by_mac' }, resource_v2a => {}, path => '/nodes/:mac/latest_locationlog' },
     { controller => 'nodes', collection_v2a => { get => 'locationlog_by_mac' }, resource_v2a => {}, path => '/nodes/:mac/locationlog' },
+
+    { controller => 'violations', resource_v2a => {}, collection_v2a => { get => 'read_list', post => 'create' }, path => '/violations' },
```

Comment:
I don't think you need those

First, read_list shouldn't be overridden (see other comment)

Then no need to say that post goes to create since it already does in the routing so you only need to override the create subroutine like you did in the controller and we should be golden

Reply:
I wonder what I'm missing. POST didn't work with this line omitted.

Reply:
I just reconfirmed that nothing works with `t/unittest/UnifiedApi/Controller/Violations.t`